src/index.ts

```diff
@@ -913,6 +934,60 @@ class ThreadPool {
 
     await Promise.all(exitEvents);
   }
+
+  async close (options : CloseOptions = kDefaultCloseOptions) {
+    this.closingUp = true;
+
+    if (options.force) {
+      const skipQueueLength = this.skipQueue.length;
+      for (let i = 0; i < skipQueueLength; i++) {
+        const taskInfo : TaskInfo = this.skipQueue.shift() as TaskInfo;
+        if (taskInfo.workerInfo === null) {
+          taskInfo.done(new AbortError());
+        } else {
+          this.skipQueue.push(taskInfo);
+        }
+      }
+
+      const taskQueueLength = this.taskQueue.size;
+      for (let i = 0; i < taskQueueLength; i++) {
+        const taskInfo : TaskInfo = this.taskQueue.shift() as TaskInfo;
+        if (taskInfo.workerInfo === null) {
+          taskInfo.done(new AbortError());
+        } else {
+          this.taskQueue.push(taskInfo);
+        }
+      }
+    }
+
+    const finishWorkersGracefully = () => new Promise<unknown[]>((resolve) => {
+      const exitEvents: Promise<unknown>[] = [];
+
+      const removeWorkerIfIsDone = (workerInfo: WorkerInfo) => {
+        if (workerInfo.taskInfos.size === 0) {
+          exitEvents.push(once(workerInfo.worker, 'exit'));
+          this._removeWorker(workerInfo);
+        }
+      };
+
+      const resolveIfWorkersIsEmpty = () => this.workers.size === 0 ? resolve(exitEvents) : undefined;
+
+      for (const workerInfo of this.workers) {
+        removeWorkerIfIsDone(workerInfo);
+        resolveIfWorkersIsEmpty();
+
+        workerInfo.port.on('message', () => {
+          removeWorkerIfIsDone(workerInfo);
+          resolveIfWorkersIsEmpty();
+        });
+      }
+    });
```

Comment:
It will be more robust if we combine `close` with `destroy`; meaning that we call `destroy` after we flush the queue or ran into a timeout